zigbeetuya/integrationpluginzigbeetuya.cpp

```diff
@@ -88,6 +108,27 @@ void IntegrationPluginZigbeeTuya::setupThing(ThingSetupInfo *info)
     ZigbeeNode *node = nodeForThing(thing);
     ZigbeeNodeEndpoint *endpoint = node->getEndpoint(0x01);
 
+
+
+    // Set the version
+    thing->setStateValue("version", endpoint->softwareBuildId());
+
+    if (thing->hasState("battery")) {
+        connectToPowerConfigurationCluster(thing, endpoint);
+    }
+
+    if (thing->thingClassId() == temperatureSensorThingClassId) {
+        qCDebug(dcZigbeeTuya()) << "Setting up temperature sensor" << thing->name() << endpoint->endpointId();;
+        connectToTemperatureMeasurementInputCluster(thing, endpoint);
+    }
+
+    if (thing->thingClassId() == htSensorThingClassId) {
+        qCDebug(dcZigbeeTuya()) << "Setting up h/t sensor" << thing->name() << endpoint->endpointId();;
+        connectToTemperatureMeasurementInputCluster(thing, endpoint);
+        connectToRelativeHumidityMeasurementInputCluster(thing, endpoint);
```

Comment:
hmm... interestingly you don't seem to use any out-of spec stuff, all of this is compliant to the ZigBee spec. Which makes me wonder why it didn't get picked up by the generic plugin already.

Seeing that your branch is based on the 1.1 release of nymea, I'm thinking that you may be running the 1.1 but support for spec-compliant H&T sensors in 1.2 and it would have just worked with the 1.2 already. Could that be the case?

Reply:
You are right. I just updated to 1.3 and now the sensor is supported by the generic Zigbee plugin. The work was for nothing. :-(  you can close the pull request.

Reply:
Hmm... dunno... IIRC the generic one would make it appear as 2 distinct devices, a Temp sensor and a Humidity sensor as the generic one just pops up a single thing for each endpoint it recognizes. Is that right? If so, there may be value in this pull request in that it makes it appear nicely as a single H&T sensor device.

Reply:
no the generic plugin does the same. 

The generic plugin, only add a humidity sensor if it finds a temperature sensor. 

I looked a bit to see if there was a humidity sensor only, without temperature. But apparently there is no such thing.

Reply:
Oh, I see. Thanks for verifying... Well, I suppose in that case, this pull request seems obsolete indeed... Sorry for that :/